### DISCUSSION TITLE
chapter_appendix/installation/

### DISCUSSION THREAD

**Discussion by giscus** [krahets/hello-algo#143] [Announcements]
# chapter_appendix/installation/

一本动画图解、能运行、可提问的数据结构与算法入门书

https://www.hello-algo.com/chapter_appendix/installation/

**Comment by twentyfourw24** [comment]
c和c++的环境不安装mingw只安装vscode或visual studio行吗

**Reply by Toona-sinensis** [reply]
vscode+mingw=vs,你直接安一个vs就行，或者安一个devc++也挺好

**Comment by ulic-youthlic** [comment]


**Reply by twentyfourw24** [reply]
可是我用vs只能查看代码却不能运行是怎么回事啊

**Comment by ulic-youthlic** [comment]


**Reply by twentyfourw24** [reply]
那把代码复制进去但是显示错误或者警告是怎么回事啊大佬

**Reply by ulic-youthlic** [reply]
给截图

**Reply by twentyfourw24** [reply]
<img width="1280" alt="image" src="https://github.com/krahets/hello-algo/assets/150130988/7452031e-dfa8-4c0b-9b95-265afcc9e601">

**Reply by ulic-youthlic** [reply]
可能你的代码没有贴全，但你的VS是正常工作的

**Reply by twentyfourw24** [reply]
啊？全部复制下来了啊。那我再看看吧。谢谢大佬

**Reply by krahets** [reply]
@twentyfourw24 需要去仓库里下载整个代码库哈

**Reply by twentyfourw24** [reply]
<img width="600" alt="image" src="https://github.com/krahets/hello-algo/assets/150130988/339cc22f-36b7-4fa1-9c91-3e8240a9b65e">

是这个吗大佬

**Reply by krahets** [reply]
对的，代码都在 codes 文件夹里～

**Comment by 98Hpp** [comment]


**Reply by Byssfz** [reply]
vscode环境的配置有什么用，刚把mingw配置好，后面不会弄了，settings在哪里不会弄，求教，本人是刚入门算法的小白一枚，不懂就问

**Reply by Byssfz** [reply]
我也找不到

**Reply by krahets** [reply]
@byssfz 可以网上搜一搜 VSCode 配置 mingw 的视频，应该蛮多的。我可能最近也会录一个～

**Reply by Byssfz** [reply]
搞好了，谢谢大佬，好像解压文件夹的时候只需要解压code的那一个就行了，其他的好像没啥用，我也看不懂哈哈

**Comment by Byssfz** [comment]
可选）打开 Settings 页面，搜索 Clang_format_fallback Style 代码格式化选项，设置为 { BasedOnStyle： Microsoft， BreakBeforeBraces： Attach } 。这个是干嘛的？我已经弄了不知道有啥用hh,小白已经懵逼了，是让软件更好用的吗？！@Krahets

**Reply by krahets** [reply]
Hi，这个是设置代码格式化风格，设置之后，可以保证使用自动代码格式化时，与仓库中的 C 和 C++ 代码保持一致。不设置也没关系的，只是在格式化代码的时候，代码格式风格会变一下

**Reply by Byssfz** [reply]
无脑与大佬保持一样的风格

**Comment by Yerosius** [comment]
希望能写出Jetbrains的IDE教程

**Reply by ulic-youthlic** [reply]
JB 的 `Python` 和 `C/C++` `IDE` 都是开箱即用的 :D
其他的没用过，不清楚。

**Comment by Water1236** [comment]
我C语言环境搭建总是失败，在教程那个配置到环境时不知道怎么搞，希望作者大大可以帮我看看

**Reply by cheng-tian-ao** [reply]
新手不知道怎么装环境可以试试用visual stdio，这个不用装环境，想用visual code可以在csdn和b站上找具体的操作办法。

**Comment by baixinran** [comment]
【1.python环境:3.（可选）在命令行输入 pip install black ，安装代码格式化工具。】
这个地方安装的时候出现了问题（win10版本：22H2；Python版本：python-3.12.2）
解决方案：在cmd窗口输入   pip3 install black
参考：https://bobbyhadz.com/blog/python-syntaxerror-invalid-syntax-pip-install

**Comment by C-rayon** [comment]
请问 tasks.json 那步出现
1、Value is not accepted. Valid values: "process".(1)[Ln 4, Col 21]
2、Unrecognized problem matcher. ls the extension that contributes this problem matcher installed? (1) [Ln 41, Col 17]
怎么解决呢？

**Comment by Gangan-Jamie** [comment]
没看懂啊，IDE和语言环境有啥区别呢，都下载了，然后呢，怎么这就开始实操了，在哪个面板写代码都没看见啊家人们啊阿啊

**Reply by Gangan-Jamie** [reply]
一点都不适合小白，差评奥，从入门到放弃

**Reply by pianfan** [reply]
不是哥们，不懂的不会自己搜？这是讲算法入门的不是编程入门奥：）

**Reply by zhuiguang49** [reply]
哥们真得给你当保姆啊

**Reply by arvinli163** [reply]
这。。。。。。。

**Reply by HighBigSky** [reply]
布什.戈门？升官发财另寻他出，编程真的不适合你

**Reply by yulong-ge** [reply]
很难想象这是GPT时代能问出来的问题

**Reply by WaterWhisperer** [reply]
STFW

**Reply by smile-1973** [reply]
这个真得搜 ，b站 搜教程就行 ，相当与 你 小学知识看高中 ，先 把编程基础打好 ，人家写的确实不错，但是这个与你 遇到困难的点不一致 ，和书 好坏没有关系

**Comment by moqsien** [comment]
想要快速配置和管理开发环境的，可以使用：https://github.com/gvcgo/version-manager

这是一款支持各种编程语言SDK版本管理的通用管理器。
官方文档：https://vdocs.vmr.us.kg/zh-cn/

**Comment by SixAnChen** [comment]
来啦来啦，膜拜膜拜大佬~

**Comment by Luna-CaiC** [comment]
大佬们，VS Code 已经下载和安装，python, c++, java环境都安装完了，请问要根据这本书学并要用source code of this book的话，后面7个语言环境一定都要安装吗？

**Reply by krahets** [reply]
Hi，不需要，只需要安装一门语言就行（你使用的编程语言）

**Reply by Luna-CaiC** [reply]
okk, got it. 非常感谢🙏
